### PR TITLE
layout: Implement computation of table column widths

### PR DESCRIPTION
**Pull request by mrobinson**
This change implements the various steps of table column width
computation, ignoring features that don't exist yet (such as separated
borders, column elements, and colgroups).

Co-authored-by: Oriol Brufau <obrufau@igalia.com>

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

### CODE REVIEW

File: tests/wpt/meta/css/CSS2/floats-clear/floats-149.xht.ini [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[floats-149.xht]
```
Comment (Loirooriol) [inline tests/wpt/meta/css/CSS2/floats-clear/floats-149.xht.ini:1]: I think this shouldn't be regressing, it's quite strange that two cells with `width: 50%` end up having a different width.

```html
<style>
table {
  width: 200px;
  border-spacing: 0px;
}
td {
  padding: 50px 0px;
  width: 50%;
  background: linear-gradient(to right, cyan, magenta);
}
</style>
<table>
  <tr><td></td><td></td></tr>
</table>
```

**Comment by mrobinson** [comment]
Oh, funny. I didn't see your fix and pushed the exact same fix. :)

File: components/layout_2020/table/layout.rs [hunk 2]
```diff
@@ -63,24 +86,473 @@ impl<'a> TableLayout<'a> {
...
+
+                let cell_measures = &self.cell_measures[resolved_coords.y][resolved_coords.x];
+                let cell_inline_content_sizes = cell_measures.content_sizes;
+
+                let columns_spanned = resolved_coords.x..resolved_coords.x + cell.colspan;
+                let baseline_content_sizes: ContentSizes = (columns_spanned.clone())
+                    .map(|spanned_column_index| {
+                        old_column_measures[spanned_column_index].content_sizes
+                    })
+                    .fold(ContentSizes::zero(), |a: ContentSizes, b| a + b);
```
Comment (Loirooriol) [inline components/layout_2020/table/layout.rs:374]: Unnecessary parentheses around `columns_spanned.clone()`. Could also merge the map and the fold, but maybe it's less clear:

```suggestion
                let baseline_content_sizes: ContentSizes = columns_spanned.clone().fold(
                    ContentSizes::zero(),
                    |acc: ContentSizes, spanned_column_index| {
                        acc + old_column_measures[spanned_column_index].content_sizes
                    },
                );
```

**Review by Loirooriol — APPROVED** [review]

**Comment by mrobinson** [comment]
Thanks for the review. I've made all of these changes.

**Pull request merged**